ui/js/models/resultsets_store.js

```diff
@@ -416,30 +405,22 @@ treeherder.factory('ThResultSetStore', [
             }
         };
 
-        /**
-         * Go through map of loaded unclassified jobs and check against current
-         * enabled tiers to get this count.
-         */
-        var updateUnclassifiedFailureCountForTiers = function () {
-            repoData.unclassifiedFailureCountForTiers = 0;
-            Object.values(repoData.unclassifiedFailureMap).forEach((job) => {
-                if (thJobFilters.isFilterSetToShow('tier', job.tier)) {
-                    repoData.unclassifiedFailureCountForTiers += 1;
-                }
-            });
-        };
-
         /**
          * Loops through the map of unlcassified failures and checks if it is
          * within the enabled tiers and if the job should be shown. This essentially
          * gives us the difference in unclassified failures and, of those jobs, the
          * ones that have been filtered out
-         *
-         * @private
          */
-        var updateFilteredUnclassifiedFailureCount = function () {
+        const recalculateUnclassifiedCounts = function () {
```

Comment:
I consolidated the two functions to get the unclassified and the "filtered" unclassified into this function.  This was more efficient, but wasn't strictly necessary for this update.  However, in this particular case, I had to manually call ``thJobFilters.refreshFilterCaches`` because the events could hit a race condition where those caches were not yet updated when we need to know what the new tiers/filters are set to.  When I convert these two models (filters and pushes) away from Angular, I will likely rely more on callbacks like this than events anyway.